itou/approvals/export.py

```diff
@@ -32,92 +32,149 @@
     "date_debut_embauche",
     "date_fin_embauche",
 ]
+
+# These values were formerly computed dynamically in the rendering loop
+# It is *way* faster to use static values to change the width of columns once
+WIDTHS_WS1 = [14, 39, 33, 14, 15, 19, 17, 11, 37, 21, 14, 73, 9, 19, 19]
```

Comment:
Toutes ces valeurs hardcodées et magiques me perturbent. C'est la longueur max de chaque champ ou bien la largeur visuelle de la colonne dans le excel? Comment savoir quel champ correspond à quelle valeur? (Peut-être fusionner `FIELDS` et `WIDTHS` en un seul `OrderedDict` plus lisible?) Si une de ces valeurs doit changer pour une prochain export, comment s'en rendra-t-on compte et comment trouvera-t-on la nouvelle valeur?

Question idiote : pourquoi ne pas juste mettre 50 partout et simplifier le code? (si c'est la longueur max de chaque champ)

Reply:
Oui effectivement, c'est un peu "out of the blue"
A l'origine, je calculais ces valeurs dynamiquement, selon la longueur de la valeur contenue par la cellule.
Mais j'ai vu que recalculer la taille des cellules à la volée ralentissait grandement la boucle principale.
J'ai pris les valeurs maximales et en ai fait des valeurs statiques, d’où l'aspect un peu magique.
Pour ce qui est de 50 partout, j'essaye et je vais voir si le résultat est potable

Reply:
Je coupe dans le code. 50 partout